besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1177,6 +1223,25 @@ private JsonRpcConfiguration jsonRpcConfiguration() {
           "Unable to authenticate JSON-RPC HTTP endpoint without a supplied credentials file or authentication public key file");
     }
 
+    if (isRpcHttpTlsEnabled) {
+      if (rpcHttpTlsKeystoreFile == null) {
+        throw new ParameterException(
+            commandLine,
+            "Key store file is required when TLS is enabled for JSON-RPC HTTP endpoint");
+      }
+
+      if (rpcHttpTlsKeystorePassword == null) {
+        throw new ParameterException(
+            commandLine,
+            "Key store password is required when TLS is enabled for JSON-RPC HTTP endpoint");
+      }
+
+      if (rpcHttpTlsTruststoreFile != null && rpcHttpTlsTruststorePassword == null) {
```

Comment:
does this mean the TlsTruststoreFile is not required? But if it is provided, then the password must exist?

ALso - should we do the opposite check too - if people specify _any_ TLS option, but not enabled TLS, then raise an error?

Reply:
yes, if trust store file is specified, then password must be specified. Otherwise we assume that user doesn't want client authentication hence no trust store will be set.

For the second point, yes, the "check" for TLS related options without TLS enabled is performed on line 1191 above.